jenkins/bootstrap_test.py

```diff
@@ -757,18 +769,30 @@ def testBootstrapPullYaml(self):
 
     def testOnlyJobs(self):
         """Ensure that everything in jobs/ is a valid job name and script."""
-        for path, _, filenames in os.walk(
-            os.path.dirname(bootstrap.job_script(JOB))):
-
-            for job in filenames:
-                # Jobs should have simple names
-                self.assertTrue(re.match(r'[0-9a-z-]+.sh', job), job)
-                job_path = os.path.join(path, job)
-                # Jobs should point to a real, executable file
-                # Note: it is easy to forget to chmod +x
-                self.assertTrue(os.path.isfile(job_path), job_path)
-                self.assertFalse(os.path.islink(job_path), job_path)
-                self.assertTrue(os.access(job_path, os.X_OK|os.R_OK), job_path)
+        for job, job_path in self.jobs:
+            # Jobs should have simple names
+            self.assertTrue(re.match(r'[0-9a-z-]+.sh', job), job)
+            # Jobs should point to a real, executable file
+            # Note: it is easy to forget to chmod +x
+            self.assertTrue(os.path.isfile(job_path), job_path)
+            self.assertFalse(os.path.islink(job_path), job_path)
+            self.assertTrue(os.access(job_path, os.X_OK|os.R_OK), job_path)
+
+    def testAllProjectAreUnique(self):
```

Comment:
I'm not a huge fan of this. What if we have a comment saying `# TODO(spxtr): set PROJECT back to such-and-such when #12345 is done`?